Rewrite networking.rst:

Mrt134 opened the pull request:
- Add PEP8.
- Add index and internal reference points.
- Add more robust Sphinx/ReST syntax/directives.
- Fix grammar, spelling, and rewrite sentences for clarity.
- Rework note and tip boxes, and add emphasis elements to important words.
- Remove some obsolete information.
To do:
- Developer review of examples for accuracy is in order.

Make sure to follow and check these boxes before submitting a PR! Thank you.

- [x] Explain the feature
- [x] Read [CONTRIBUTING.md](https://github.com/iocage/iocage/blob/master/CONTRIBUTING.md)

Mrt134 committed:
Rewrite networking.rst:

- Add PEP8.
- Add index and internal reference points.
- Add more robust Sphinx/ReST syntax/directives.
- Fix grammar, spelling, and rewrite sentences for clarity.
- Rework note and tip boxes, and add emphasis elements to important words.
- Remove some obsolete information.
To do:
- Developer review of examples for accuracy is in order.

skarekrow commented on doc/source/networking.rst:
```diff
@@ -1,185 +1,215 @@
 **Verify visible IP configuration in the jail**
 
-*(jail must be running for this to work)*
+.. note:: The jail **must** be running for this example to work.
```
We can drop this note, we start jails automatically now.

skarekrow commented on doc/source/networking.rst:
```diff
@@ -1,185 +1,215 @@
-  ifconfig_bridge0="addm em0 up"
-  ifconfig_em0="up"
+   # plumb interface em0 into bridge0
+   ifconfig_bridge0="addm em0 up"
```
Some not about replacing `em0` with their network adapter should probably be here and everywhere else it's mentioned.

skarekrow reviewed (changes requested):
Thanks :D

Mrt134 committed:
Update networking.rst

Address review comments. Added a single warning box about "em0" to the initial section, which looks sufficient in build testing.
